### DISCUSSION TITLE
ESMF Installation issue

### DISCUSSION THREAD

**Discussion by gaohongyin** [NASA-LIS/LISF#1507] [Q&A]
Hi everyone,
I am trying to install LISF on my workstation with Ubuntu 18.04.
When installing the ESMF library, I can successfully "make all", but when type "make install", it always report error saying

makefile:668: recipe for target 'install' failed
make: *** [install] Error 1

Here is my setting for ESMF:
export ESMF_DIR=/usr/local/esmf/8.1.1/esmf-ESMF_8_1_1
export ESMF_INSTALL_PREFIX=/usr/local/esmf/8.1.1/esmf-ESMF_8_1_1
export ESMF_COMM=mpiuni
export ESMF_COMPILER=gfortran
export ESMF_ABI=64
export ESMF_NETCDF_INCLUDE=/usr/local/netcdf/4.7.4/include
export ESMF_NETCDF_LIBPATH=/usr/local/netcdf/4.7.4/lib
export ESMF_NETCDF_LIBS="-lnetcdf -lnetcdff"
export ESMF_CXXCOMPILER=g++
export ESMF_CXXLINKER=g++
export ESMF_F90COMPILER=gfortran
export ESMF_F90LINKER=gfortran

I've been struggling with this for a long time, I appreciate if anyone can provide help!

![Uploading Screenshot from 2024-03-21 18-37-29.png…]()

**Comment by jvgeiger** [comment]
Hello.  I am unable to see the screenshot that you uploaded.  Would you please try again?  Or post more of the error message that you get?

**Reply by gaohongyin** [reply]
Thank you for your reply. When I tried "make check", I got errors as follows:

make[10]: Entering directory '/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1/mod/modO/Linux.gfortran.64.openmpi.default'
gfortran -c -O -fPIC  -m64 -mcmodel=small -pthread -ffree-line-length-none  -fopenmp -I/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1/src/Infrastructure/IO/PIO -I/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1/build_config/Linux.gfortran.default -I/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1/src/Infrastructure -I/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1/src/Superstructure -I/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1/mod/modO/Linux.gfortran.64.openmpi.default -I/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1/src/include   -DNO_MPIMOD -DESMF_NO_INTEGER_1_BYTE -DESMF_NO_INTEGER_2_BYTE -DESMF_MOAB=1 -DESMF_LAPACK=1 -DESMF_NO_ACC_SOFTWARE_STACK=1 -DESMF_YAMLCPP=1 -DESMF_YAML=1 -DESMF_PIO=1 -DESMF_MPIIO -DESMF_NO_OPENACC -DESMF_BOPT_O -DESMF_TESTCOMPTUNNEL -DSx86_64_small=1 -DESMF_OS_Linux=1 -DESMF_COMM=openmpi -DESMF_DIR=/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1 /home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1/src/Infrastructure/IO/PIO/pio_kinds.F90 -o /home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1/obj/objO/Linux.gfortran.64.openmpi.default/pio_kinds.o
/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1/src/Infrastructure/IO/PIO/pio_kinds.F90:31: Error: Can't open included file 'mpif.h'
/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1/build/common.mk:3718: recipe for target '/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1/obj/objO/Linux.gfortran.64.openmpi.default/pio_kinds.o' failed
make[10]: *** [/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1/obj/objO/Linux.gfortran.64.openmpi.default/pio_kinds.o] Error 1
make[10]: Leaving directory '/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1/mod/modO/Linux.gfortran.64.openmpi.default'
/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1/build/common.mk:2173: recipe for target 'esmflib' failed
make[9]: *** [esmflib] Error 2
make[9]: Leaving directory '/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1/mod/modO/Linux.gfortran.64.openmpi.default'
/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1/build/common.mk:2169: recipe for target 'tree_lib' failed
make[8]: *** [tree_lib] Error 2
make[8]: Leaving directory '/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1/src/Infrastructure/IO/PIO'
/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1/build/common.mk:3661: recipe for target 'tree' failed
make[7]: *** [tree] Error 1
make[7]: Leaving directory '/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1/src/Infrastructure/IO'
/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1/build/common.mk:3661: recipe for target 'tree' failed
make[6]: *** [tree] Error 1
make[6]: Leaving directory '/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1/src/Infrastructure'
/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1/build/common.mk:3661: recipe for target 'tree' failed
make[5]: *** [tree] Error 1
make[5]: Leaving directory '/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1/src'
/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1/build/common.mk:3661: recipe for target 'tree' failed
make[4]: *** [tree] Error 1
make[4]: Leaving directory '/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1'
/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1/build/common.mk:2136: recipe for target 'build_libs' failed
make[3]: *** [build_libs] Error 2
make[3]: Leaving directory '/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1'
/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1/build/common.mk:2128: recipe for target 'lib' failed
make[2]: *** [lib] Error 2
make[2]: Leaving directory '/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1'
/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1/build/common.mk:2079: recipe for target 'reqfile_libesmf' failed
make[1]: *** [reqfile_libesmf] Error 2
make[1]: Leaving directory '/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1'
/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1/build/common.mk:2334: recipe for target 'check' failed
make: *** [check] Error 2


my make information is:
ESMF_VERSION_STRING:    8.1.1
Not a Git repository
--------------------------------------------------------------
 
--------------------------------------------------------------
Make version:
GNU Make 4.1
Built for x86_64-pc-linux-gnu
Copyright (C) 1988-2014 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

--------------------------------------------------------------
Fortran Compiler version:
GNU Fortran (Ubuntu 11.4.0-2ubuntu1~18.04) 11.4.0
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.


--------------------------------------------------------------
C++ Compiler version:
g++ (Ubuntu 11.4.0-2ubuntu1~18.04) 11.4.0
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.


--------------------------------------------------------------
Preprocessor version:
gcc (Ubuntu 11.4.0-2ubuntu1~18.04) 11.4.0
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

 
--------------------------------------------------------------
 * User set ESMF environment variables *
ESMF_ABI=64
ESMF_COMM=openmpi
ESMF_COMPILER=gfortran
ESMF_CXXCOMPILER=g++
ESMF_CXXLINKER=g++
ESMF_DIR=/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1
ESMF_F90COMPILER=gfortran
ESMF_F90LINKER=gfortran
ESMF_INSTALL_PREFIX=/usr/local/esmf-8.1.1
ESMF_LAPACK=netlib
ESMF_NETCDF_INCLUDE=/usr/local/netcdf4/include
ESMF_NETCDF_LIBPATH=/usr/local/netcdf4/lib
ESMF_NETCDF_LIBS=-lnetcdf -lnetcdff
ESMF_NETCF_LIBS=-lnetcdf -lnetcdff
PWD=/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1
 
--------------------------------------------------------------
 * ESMF environment variables *
ESMF_DIR: /home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1
ESMF_OS:                Linux
ESMF_MACHINE:           x86_64
ESMF_ABI:               64
ESMF_COMPILER:          gfortran
ESMF_BOPT:              O
ESMF_COMM:              openmpi
ESMF_SITE:              default
ESMF_PTHREADS:          ON
ESMF_OPENMP:            ON
ESMF_OPENACC:           OFF
ESMF_CXXSTD:            11
ESMF_ARRAY_LITE:        FALSE
ESMF_NO_INTEGER_1_BYTE: TRUE
ESMF_NO_INTEGER_2_BYTE: TRUE
ESMF_FORTRANSYMBOLS:    default
ESMF_MAPPER_BUILD:      OFF
ESMF_AUTO_LIB_BUILD:    ON
ESMF_DEFER_LIB_BUILD:   ON
ESMF_SHARED_LIB_BUILD:  ON
ESMF_TRACE_LIB_BUILD:   ON
ESMF_TESTEXHAUSTIVE:    OFF
ESMF_TESTCOMPTUNNEL:    ON
ESMF_TESTWITHTHREADS:   OFF
ESMF_TESTMPMD:          OFF
ESMF_TESTSHAREDOBJ:     OFF
ESMF_TESTFORCEOPENMP:   OFF
ESMF_TESTFORCEOPENACC:  OFF
ESMF_TESTHARNESS_ARRAY: default
ESMF_TESTHARNESS_FIELD: default
ESMF_MPIRUN:            mpirun 
 
--------------------------------------------------------------
 * ESMF environment variables pointing to 3rd party software *
ESMF_MOAB:               internal
ESMF_LAPACK:             netlib
ESMF_LAPACK_LIBS:        -llapack -lblas
ESMF_ACC_SOFTWARE_STACK: none
ESMF_PIO:                internal
ESMF_YAMLCPP:            internal
 
--------------------------------------------------------------
 * ESMF environment variables for final installation *
ESMF_INSTALL_PREFIX:    /usr/local/esmf-8.1.1
ESMF_INSTALL_HEADERDIR: include
ESMF_INSTALL_MODDIR:    mod/modO/Linux.gfortran.64.openmpi.default
ESMF_INSTALL_LIBDIR:    lib/libO/Linux.gfortran.64.openmpi.default
ESMF_INSTALL_BINDIR:    bin/binO/Linux.gfortran.64.openmpi.default
ESMF_INSTALL_DOCDIR:    doc
 
--------------------------------------------------------------
 * ESMF Benchmark directory and parameters *
ESMF_BENCHMARK_PREFIX:         ./DEFAULTBENCHMARKDIR
ESMF_BENCHMARK_TOLERANCE:      20%
ESMF_BENCHMARK_THRESHOLD_MSEC: 500
 
--------------------------------------------------------------
 * Other relevant environment variables *
PATH:    /usr/local/geotiff/bin:/usr/local/proj/bin:/usr/local/sqlite3/bin:/usr/local/eccodes/bin:/usr/local/cmake-3.23.0/bin:/usr/local/openjpeg/bin:/usr/local/openmpi/bin:/home/gyin/anaconda3/bin:/home/gyin/anaconda3/condabin:/usr/local/netcdf4/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
LD_LIBRARY_PATH: /usr/local/geotiff/lib:/usr/local/eccodes/lib:/usr/local/openjpeg/lib:/usr/local/openmpi/lib:/usr/local/netcdf4/lib:
 
--------------------------------------------------------------
 * Compilers, Linkers, Flags, and Libraries *
Location of the preprocessor:      /usr/bin/gcc
Location of the Fortran compiler:  /usr/bin/gfortran
Location of the Fortran linker:    /usr/bin/gfortran
Location of the C++ compiler:      /usr/bin/g++
Location of the C++ linker:        /usr/bin/g++

Fortran compiler flags:
ESMF_F90COMPILER: gfortran
ESMF_F90COMPILEOPTS: -O -fPIC  -m64 -mcmodel=small -pthread -ffree-line-length-none  -fopenmp
ESMF_F90COMPILEPATHS: -I/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1/mod/modO/Linux.gfortran.64.openmpi.default -I/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1/src/include 
ESMF_F90COMPILECPPFLAGS: -DESMF_NO_INTEGER_1_BYTE -DESMF_NO_INTEGER_2_BYTE -DESMF_MOAB=1 -DESMF_LAPACK=1 -DESMF_NO_ACC_SOFTWARE_STACK=1 -DESMF_YAMLCPP=1 -DESMF_YAML=1 -DESMF_PIO=1 -DESMF_MPIIO -DESMF_NO_OPENACC -DESMF_BOPT_O -DESMF_TESTCOMPTUNNEL -DSx86_64_small=1 -DESMF_OS_Linux=1 -DESMF_COMM=openmpi -DESMF_DIR=/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1
ESMF_F90COMPILEFREECPP: 
ESMF_F90COMPILEFREENOCPP: -ffree-form
ESMF_F90COMPILEFIXCPP: -cpp -ffixed-form
ESMF_F90COMPILEFIXNOCPP: 

Fortran linker flags:
ESMF_F90LINKOPTS:    -m64 -mcmodel=small -pthread -Wl,--no-as-needed  -fopenmp
ESMF_F90LINKPATHS: -L/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1/lib/libO/Linux.gfortran.64.openmpi.default  -L/usr/lib/gcc/x86_64-linux-gnu/11/
ESMF_F90LINKRPATHS: -Wl,-rpath,/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1/lib/libO/Linux.gfortran.64.openmpi.default  -Wl,-rpath,/usr/lib/gcc/x86_64-linux-gnu/11/
ESMF_F90LINKLIBS:   -lrt -lstdc++ -ldl -llapack -lblas
ESMF_F90ESMFLINKLIBS: -lesmf   -lrt -lstdc++ -ldl -llapack -lblas

C++ compiler flags:
ESMF_CXXCOMPILER: g++
ESMF_CXXCOMPILEOPTS: -std=c++11 -O -DNDEBUG -fPIC -DESMF_LOWERCASE_SINGLEUNDERSCORE -m64 -mcmodel=small -pthread  -fopenmp
ESMF_CXXCOMPILEPATHS:  -I/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1/src/include   -I/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1/src/prologue/yaml-cpp/include
ESMF_CXXCOMPILECPPFLAGS: -DESMF_NO_INTEGER_1_BYTE -DESMF_NO_INTEGER_2_BYTE -DESMF_MOAB=1 -DESMF_LAPACK=1 -DESMF_NO_ACC_SOFTWARE_STACK=1 -DESMF_YAMLCPP=1 -DESMF_YAML=1 -DESMF_PIO=1 -DESMF_MPIIO -DESMF_NO_OPENACC -DESMF_BOPT_O -DESMF_TESTCOMPTUNNEL -DSx86_64_small=1 -DESMF_OS_Linux=1 -DESMF_COMM=openmpi -DESMF_DIR=/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1 -D__SDIR__='' -DESMF_CXXSTD=11 -DESMF_NO_SIGUSR2

C++ linker flags:
ESMF_CXXLINKOPTS:   -m64 -mcmodel=small -pthread -Wl,--no-as-needed  -fopenmp
ESMF_CXXLINKPATHS: -L/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1/lib/libO/Linux.gfortran.64.openmpi.default  -L/usr/lib/gcc/x86_64-linux-gnu/11/
ESMF_CXXLINKRPATHS: -Wl,-rpath,/home/gyin/Documents/LIS/dependencies/esmf/esmf-ESMF_8_1_1/lib/libO/Linux.gfortran.64.openmpi.default  -Wl,-rpath,/usr/lib/gcc/x86_64-linux-gnu/11/
ESMF_CXXLINKLIBS:  -lmpi_mpifh -lrt -lgfortran -ldl -llapack -lblas
ESMF_CXXESMFLINKLIBS: -lesmf  -lmpi_mpifh -lrt -lgfortran -ldl -llapack -lblas

Shared library build:
ESMF_SL_LIBS_TO_MAKE: libesmf
ESMF_SL_SUFFIX:       so
ESMF_SL_LIBLINKER:    g++
ESMF_SL_LIBOPTS:       -shared  -fopenmp
ESMF_SL_LIBLIBS:
